utils/network.go

```diff
@@ -62,23 +140,23 @@ func (n *Network) MarshalJSON() ([]byte, error) {
 }
 
 func (n *Network) Set(s string) error {
-	switch s {
-	case "MAINNET", "mainnet":
-		*n = Mainnet
-	case "GOERLI", "goerli":
-		*n = Goerli
-	case "GOERLI2", "goerli2":
-		*n = Goerli2
-	case "INTEGRATION", "integration":
-		*n = Integration
-	case "SEPOLIA", "sepolia":
-		*n = Sepolia
-	case "SEPOLIA_INTEGRATION", "sepolia-integration":
-		*n = SepoliaIntegration
-	default:
-		return ErrUnknownNetwork
+	sLower := strings.ToLower(s)
+
+	predefinedNetworks := map[string]Network{
+		"mainnet":             Mainnet,
+		"goerli":              Goerli,
+		"goerli2":             Goerli2,
+		"integration":         Integration,
+		"sepolia":             Sepolia,
+		"sepolia-integration": SepoliaIntegration,
+		"custom":              CustomNetwork,
+	}
+
+	if network, ok := predefinedNetworks[sLower]; ok {
```

Comment:
```suggestion
	if network, ok := predefinedNetworks[strings.ToLower(s)]; ok {
```
nit: we can remove the `sLower` variable.